src/state/trade/reducer.ts

```diff
@@ -0,0 +1,53 @@
+import { createReducer } from "@reduxjs/toolkit";
+import {
+  DefaultTxnSettings,
+  typeInput,
+  selectLeverage,
+  selectPositionSide,
+  setSlippage,
+  setTxnDeadline,
+  resetTradeState
+} from "./actions";
+
+export interface TradeState {
+  readonly typedValue: string | undefined;
```

Comment:
```suggestion
  readonly typedValue: string;
```
can you remove the undefined type, and handle the undefined case separately?